speech_api/Dockerfile

```diff
@@ -0,0 +1,42 @@
+FROM python:3.10-slim-buster
+LABEL maintainer="IDinsight"
+
+ENV DEBIAN_FRONTEND=noninteractive
+ARG NAME=speech_service
+ARG HOME_DIR=/usr/src/${NAME}
+ARG PORT=8001
+ARG WHISPER_MODEL_DIR=/whisper_models
+
+RUN apt-get update && apt-get install -y \
+    ffmpeg dos2unix \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/*
+
+RUN useradd -ms /bin/bash -d /usr/src/${NAME} container_user
+
+RUN mkdir -p ${WHISPER_MODEL_DIR} \
+    && chown -R container_user:container_user ${WHISPER_MODEL_DIR}
+
+COPY requirements.txt /requirements.txt
+RUN pip install --no-cache-dir -r /requirements.txt --extra-index-url https://download.pytorch.org/whl/cpu
+
+WORKDIR ${HOME_DIR}
+
+RUN python -c "import os, whisper; whisper.load_model('base', download_root='${WHISPER_MODEL_DIR}')"
```

Comment:
Should we skip this line? If we are not using base?

Reply:
Oh, OR, should we download the model according to a build argument `PREFERRED_MODEL`?

For simplicity though we can just stick to `small` for now and parametrize this later if need be..

Reply:
i actually wanted to download the model according to the build argument `PREFERRED_MODEL` originally, but i dont know it wasnt exporting the env variable properly in the dockerfile so i could use it. what is the proper syntax for it? cause it was giving me issues

Reply:
You can refer to the syntax here. We can explicitly pass the build argument and tag the image accordingly!  `-t idinsight/speeach-service:latest-${PREFERRED_WHISPER_MODEL}` or something?

https://github.com/IDinsight/ask-a-question/blob/main/.github/workflows/deploy_gcp_admin_app.yaml#L55-L60